src/libsystemd-network/dhcp6-option.c

```diff
@@ -167,6 +168,46 @@ int dhcp6_option_append_fqdn(uint8_t **buf, size_t *buflen, const char *fqdn) {
         return r;
 }
 
+int dhcp6_option_append_vendor_class(uint8_t **buf, size_t *buflen, char **vendor_class) {
+        _cleanup_free_ uint8_t *p = NULL;
+        uint32_t enterprise_identifier;
+        size_t total, offset;
+        char **s;
+
+        assert(buf);
+        assert(*buf);
+        assert(buflen);
+        assert(vendor_class);
+
+        unaligned_write_be32(&enterprise_identifier, SYSTEMD_PEN);
```

Comment:
the variable `enterprise_identifier` is aligned here, you just declared it above, hence you can just do:

```
enterprise_identifier = htobe32(SYSTEMD_PEN);
```